globus_sdk/auth/client_types/confidential_client.py

```diff
@@ -72,6 +74,9 @@ def oauth2_client_credentials_tokens(self, requested_scopes=None):
         """
         self.logger.info('Fetching token(s) using client credentials')
         requested_scopes = requested_scopes or DEFAULT_REQUESTED_SCOPES
+        # convert scopes iterable to string immediately on load
+        if not isinstance(requested_scopes, six.string_types):
```

Comment:
Consider `hasattr(requested_scopes, '__iter__')` for duck-typing, as opposed to `isinstance`-checking.